### ISSUE TITLE
Drawn layer is not the top layer on the map

### ISSUE THREAD

**Issue by mem48** [SDCA-tool/sdca-website#32]
![image](https://user-images.githubusercontent.com/15819577/173810023-2d02de90-d510-4ce3-b25a-7efe6b4a0e44.png)

When you add data layers to the map and then draw on an intervention, the intervention appears behind the data layers. Anything you draw should always be the top layer.

![image](https://user-images.githubusercontent.com/15819577/173810288-8bb3b3ad-5358-408e-8562-d469c0a58a5e.png)

Also when clicking on an intervention you trigger multiple popups.

![image](https://user-images.githubusercontent.com/15819577/173810465-d5cc9ad2-c0ee-4df3-bcb1-8cf335b7f241.png)

I suggest that you only show the popup for the top layer.

Finally, popups still trigger when drawing lines. I suggest disabling popups while drawing is taking place

![image](https://user-images.githubusercontent.com/15819577/173810830-1d37f392-727e-4646-8205-c548b6423d82.png)

**Comment by mvl22** [comment]
> Finally, popups still trigger when drawing lines. I suggest disabling popups while drawing is taking place

Fixed in 97fe395db4b70ea2637d6ec7f47805f6a5baa286 capturing upstream.

**Comment by mvl22** [comment]
> Don't know if this is related but the dashed lines have disappeared during the drawing stage, only the most recent dots show.

No, this was not related, but due to a dependency upgrade issue. See #34.

**Comment by mvl22** [comment]
> When you add data layers to the map and then draw on an intervention, the intervention appears behind the data layers.

This is (mainly) resolved in 680adcd8a21d363901995e3b05a44cabaa9e5e6e.

However, if you add a layer subsequently, the moving back to the top doesn't work. I'm fairly convinced this is a Mapbox GL JS bug as I'm calling the code as per the documentation. I'll leave open this ticket and try to get a minimised testcase in due course so that a bug can be reported on their repo.

**Comment by mvl22** [comment]
> Anything you draw should always be the top layer.

> However, if you add a layer subsequently, the moving back to the top doesn't work.

This is now resolved also, in 77f87cc1b613108c8ccfe1bf4f3fa52dc07571b1.

The problem here was that when layers were added, they always get added to the top. So this and the previous commit add in handling such that when a layer change happens, the foreground layers (drawing control, drawn items, could be other things in future) are explicitly shunted back to the top. Unfortunately Mapbox GL JS doesn't seem to allow an order to be explicitly specified at the point of making a layer, so this kind of workaroundy stuff seems to be necessary, unlike with Leaflet which just does the right thing naturally.

**Comment by mvl22** [comment]
> Finally, popups still trigger when drawing lines. I suggest disabling popups while drawing is taking place

This main behaviour was fixed a while back, but I see there is still a bug that when drawing a second intervention, the popup disabling doesn't seem to trigger. I'll look at this soon.

**Comment by mvl22** [comment]
The fix 34b86c960d6ddbc3a7bb1812316eb2d71de25904 in #42 to deal with basemaps being lost exposes another bug here that if the basemap layer is changed, the drawn items before that are not shown.

This will likely be the same class of bug as the above, so is best to resolve at the same time.

Workaround for now is to change the basemap first and only then start drawing.
